### PR TITLE
adding birthday check(#86)

### PR DESCRIPTION
**Pull request by acrc2**

### CODE REVIEW

File: vitaes-webapp/src/Builder.js [hunk 1]
```diff
@@ -314,6 +314,13 @@ class Builder extends Component {
         alert("Empty name field")
         return
       }
+      if(!this.props.cv.CvHeaderItem.birthday==""){
```
Comment (Arthurlpgc) [inline vitaes-webapp/src/Builder.js]: Nit: add a blank line before this if

File: vitaes-webapp/src/Builder.js [hunk 2]
```diff
@@ -314,6 +314,13 @@ class Builder extends Component {
...
+        if(!this.props.cv.CvHeaderItem.birthday.match(/^\d{4}-\d{2}-\d{2}$/)){
+          alert("Wrong birthday date format")
+          return
+        }
+      }
+
```
Comment (Arthurlpgc) [inline vitaes-webapp/src/Builder.js]: Nit remove this blank line

File: vitaes-webapp/src/Builder.js (hunk 1 above)
Comment (Arthurlpgc) [reply vitaes-webapp/src/Builder.js]: And add spaces around the equal sign

File: vitaes-webapp/src/Builder.js [hunk 3]
```diff
@@ -314,6 +314,13 @@ class Builder extends Component {
         alert("Empty name field")
         return
       }
+      if(!this.props.cv.CvHeaderItem.birthday==""){
+        if(!this.props.cv.CvHeaderItem.birthday.match(/^\d{4}-\d{2}-\d{2}$/)){
```
Comment (magsouza) [inline vitaes-webapp/src/Builder.js]: You should consider date format includes also months going from 01 to 12 and days going from 01 to 31 (or 30 or 29 or 28). Have a look on this article about date validation: https://medium.com/@esganzerla/simple-date-validation-with-javascript-caea0f71883c

File: vitaes-webapp/src/Builder.js [hunk 4]
```diff
@@ -314,6 +314,19 @@ class Builder extends Component {
         alert("Empty name field")
         return
       }
+      if (!this.props.cv.CvHeaderItem.birthday ==""){
+        const date = new Date(this.props.cv.CvHeaderItem.birthday)
+        const day = Number(this.props.cv.CvHeaderItem.birthday[8]+this.props.cv.CvHeaderItem.birthday[9])
+        const month = Number(this.props.cv.CvHeaderItem.birthday[5]+this.props.cv.CvHeaderItem.birthday[6])
+        const year = Number(this.props.cv.CvHeaderItem.birthday[0]+this.props.cv.CvHeaderItem.birthday[1]+this.props.cv.CvHeaderItem.birthday[2]+this.props.cv.CvHeaderItem.birthday[3])
+        if(!this.props.cv.CvHeaderItem.birthday.match(/^\d{4}-\d{2}-\d{2}$/)){
```
Comment (Arthurlpgc) [inline vitaes-webapp/src/Builder.js]: this should be a different if, they shouldn't be nested

File: vitaes-webapp/src/Builder.js [hunk 5]
```diff
@@ -314,6 +314,19 @@ class Builder extends Component {
         alert("Empty name field")
         return
       }
+      if (!this.props.cv.CvHeaderItem.birthday ==""){
```
Comment (Arthurlpgc) [inline vitaes-webapp/src/Builder.js]: nit: space after ==

Comment (Arthurlpgc) [reply vitaes-webapp/src/Builder.js]: And add a blank line before this if please

File: vitaes-webapp/src/Builder.js (hunk 4 above)
Comment (Arthurlpgc) [reply vitaes-webapp/src/Builder.js]: Actually validate date can be a function

File: vitaes-webapp/src/Builder.js [hunk 6]
```diff
@@ -295,6 +295,18 @@ class Builder extends Component {
         return re.test(email);
     }
 
+    validateBirthday(birthday){
```
Comment (Arthurlpgc) [inline vitaes-webapp/src/Builder.js]: rename to date, to make it more general, other than that LGTM

**Comment by Arthurlpgc** [comment]
Thanks for the great work!

**Pull request merged**